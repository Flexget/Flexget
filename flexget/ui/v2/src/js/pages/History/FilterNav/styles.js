import styled from 'react-emotion';
import Icon from 'material-ui/Icon';
import { FormControl } from 'material-ui/Form';
import theme from 'theme';

export const RotatingIcon = styled(Icon)`
  transition: ${theme.transitions.create()};
  transform: ${({ rotate }) => rotate && 'rotate(180deg)'};
`;

export const PaddedFormControl = styled(FormControl)`
  margin: 0.5rem 2rem 0.5rem;
`;
