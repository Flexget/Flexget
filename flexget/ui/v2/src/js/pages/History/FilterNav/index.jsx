import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from 'material-ui/IconButton';
import Select from 'material-ui/Select';
import Input, { InputLabel } from 'material-ui/Input';
import { MenuItem } from 'material-ui/Menu';
import SecondaryNav from 'common/SecondaryNav';
import { Spacer } from 'common/styles';
import { RotatingIcon, PaddedFormControl } from './styles';

const ENTER_KEY = 13;
const groupByFields = [
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
];

const sortByFields = [
  {
    value: 'details',
    label: 'Details',
  },
  {
    value: 'filename',
    label: 'Filename',
  },
  {
    value: 'id',
    label: 'ID',
  },
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
  {
    value: 'title',
    label: 'Title',
  },
  {
    value: 'url',
    label: 'URL',
  },
];

class FilterNav extends Component {
  static propTypes = {
    order: PropTypes.oneOf(['asc', 'desc']).isRequired,
    grouping: PropTypes.oneOf(groupByFields.map(({ value }) => value)).isRequired,
    sort: PropTypes.oneOf(sortByFields.map(({ value }) => value)).isRequired,
    toggleOrder: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  state = {
    task: '',
  }

  handleChange = (event) => {
    this.setState({ task: event.target.value });
  }

  handleKeyPress = (event) => {
    if (event.which === ENTER_KEY) {
      this.props.handleChange('task')(event);
    }
  }

  render() {
    const {
      order,
      grouping,
      sort,
      toggleOrder,
      handleChange,
    } = this.props;

    const {
      task,
    } = this.state;

    return (
      <SecondaryNav>
        <PaddedFormControl>
          <InputLabel htmlFor="task-filter">
            Filter By Task
          </InputLabel>
          <Input
            id="task-filter"
            placeholder="Task Name"
            value={task}
            onChange={this.handleChange}
            onKeyPress={this.handleKeyPress}
          />
        </PaddedFormControl>
        <Spacer />
        <PaddedFormControl>
          <InputLabel htmlFor="sort-by">Sort By</InputLabel>
          <Select
            value={sort}
            onChange={handleChange('sort')}
            input={<Input id="sort-by" />}
          >
            {sortByFields.map(({ value, label }) => (
              <MenuItem value={value} key={value}>{label}</MenuItem>
            ))}
          </Select>
        </PaddedFormControl>
        <PaddedFormControl>
          <InputLabel htmlFor="group-by">Group By</InputLabel>
          <Select
            value={grouping}
            onChange={handleChange('grouping')}
            input={<Input id="group-by" />}
          >
            {groupByFields.map(({ value, label }) => (
              <MenuItem value={value} key={value}>{label}</MenuItem>
            ))}
          </Select>
        </PaddedFormControl>
        <IconButton color="inherit" onClick={toggleOrder}>
          <RotatingIcon rotate={order === 'desc'} className="fa fa-chevron-up" />
        </IconButton>
      </SecondaryNav>
    );
  }
}

export default FilterNav;
