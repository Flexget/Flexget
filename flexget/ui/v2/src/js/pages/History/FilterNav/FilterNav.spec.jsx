import React from 'react';
import renderer from 'react-test-renderer';
import FilterNav from 'pages/History/FilterNav';

describe('pages/History/FilterNav', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FilterNav
        handleChange={jest.fn()}
        toggleOrder={jest.fn()}
        grouping="time"
        sort="time"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
