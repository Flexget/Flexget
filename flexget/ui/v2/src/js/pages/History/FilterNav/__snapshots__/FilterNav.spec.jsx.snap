// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/History/FilterNav renders correctly 1`] = `
.glamor-6 {
  position: static;
}

.glamor-5 {
  background-color: #455a64;
  min-height: 5rem;
  color: rgba(255,255,255,1);
}

.glamor-0 {
  margin: 0.5rem 2rem 0.5rem;
}

.glamor-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-4 {
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

<header
  className="MuiPaper-root-8 MuiPaper-shadow4-14 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-6 mui-fixed glamor-6"
>
  <div
    className="MuiToolbar-root-35 MuiToolbar-gutters-36 glamor-5"
  >
    <div
      className="MuiFormControl-root-37 glamor-0"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-41 MuiInputLabel-formControl-42 MuiInputLabel-animated-45"
        data-shrink={false}
        htmlFor="task-filter"
      >
        Filter By Task
      </label>
      <div
        className="MuiInput-root-51 MuiInput-formControl-52 MuiInput-inkbar-53 MuiInput-underline-59"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
      >
        <input
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-55 MuiInput-inputSingleline-62"
          defaultValue={undefined}
          disabled={false}
          id="task-filter"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Task Name"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="glamor-1"
    />
    <div
      className="MuiFormControl-root-37 glamor-0"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-41 MuiInputLabel-formControl-42 MuiInputLabel-animated-45 MuiInputLabel-shrink-44"
        data-shrink={true}
        htmlFor="sort-by"
      >
        Sort By
      </label>
      <div
        className="MuiInput-root-51 MuiInput-formControl-52 MuiInput-inkbar-53 MuiInput-underline-59"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-66"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-67 MuiSelect-selectMenu-68 MuiInput-input-55 MuiInput-inputSingleline-62"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={undefined}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Time
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="sort-by"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="hidden"
            value="time"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-71 MuiSelect-icon-70"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-37 glamor-0"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-47 MuiInputLabel-root-41 MuiInputLabel-formControl-42 MuiInputLabel-animated-45 MuiInputLabel-shrink-44"
        data-shrink={true}
        htmlFor="group-by"
      >
        Group By
      </label>
      <div
        className="MuiInput-root-51 MuiInput-formControl-52 MuiInput-inkbar-53 MuiInput-underline-59"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-66"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-67 MuiSelect-selectMenu-68 MuiInput-input-55 MuiInput-inputSingleline-62"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={undefined}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Time
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="group-by"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="hidden"
            value="time"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-71 MuiSelect-icon-70"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root-85 MuiIconButton-root-76 MuiIconButton-colorInherit-80"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label-82"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-87 fa fa-chevron-up glamor-4"
          rotate={false}
        />
      </span>
      <span
        className="MuiTouchRipple-root-94"
      />
    </button>
  </div>
</header>
`;
